src/main/java/com/pinterest/secor/io/impl/AvroParquetFileReaderWriterFactory.java

```diff
@@ -66,6 +71,21 @@ public FileWriter BuildFileWriter(LogFilePath logFilePath, CompressionCodec code
         return serialized.array();
     }
 
+    protected Schema getSchema(String topic) {
+        Schema schema;
+        if (schemaSubjectOverride != null && !schemaSubjectOverride.isEmpty()) {
+            schema = schemaRegistryClient.getSchema(schemaSubjectOverride);
+        } else {
+            try {
+                schema = schemaRegistryClient.getSchema(topic);
```

Comment:
Is this call expensive?  Does it involve remote RPC?  this schemaRegistryClient.getSchema() was happening multiple times in this routine.

Reply:
getSchema() is only called on reader/writer creation. And this is done on purpose to allow for lookup of the original subject first, and if it doesn't find this then it will lookup the subject with "-value" appended, which is the default for certain schema registry clients like Python.

I believe this is the desired procedure for schema lookup.

Reply:
Also, it appears the project is using `CachedSchemaRegistryClient` so it should cache previous requests if it is created again.

Reply:
Or, if you want I can make the default null and then append the suffix only if it is set in config. This is more explicit, but will make it less likely to work "out of the box"